pkg/status/status.go

```diff
@@ -1,65 +1,89 @@
 package status
 
 import (
-	"context"
+	"fmt"
+	"sync"
+	"time"
 
 	configv1 "github.com/openshift/api/config/v1"
 	configclient "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
 	cohelpers "github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
 	operatorhelpers "github.com/openshift/library-go/pkg/operator/v1helpers"
 	log "github.com/sirupsen/logrus"
-	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
-const clusterOperatorName = "marketplace-operator"
+const (
+	clusterOperatorName     = "marketplace-operator"
+	minSyncsBeforeReporting = 4
+	// At this point, a failure is critical so all syncs must be successful
+	// to report a working operator
+	successRatio           = 1.0
+	maxSyncsBeforeTruncate = 10000
```

Comment:
Can we document all of these constant magic numbers? I understand them now but I will probably forget if I look months from now :)

Reply:
Great point, will add comments